packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -70,31 +107,33 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     /// @notice Constructs the SystemConfig contract. Cannot set
     ///         the owner to `address(0)` due to the Ownable contract's
     ///         implementation, so set it to `address(0xdEaD)`
-    /// @param _owner             Initial owner of the contract.
-    /// @param _overhead          Initial overhead value.
-    /// @param _scalar            Initial scalar value.
-    /// @param _batcherHash       Initial batcher hash.
-    /// @param _gasLimit          Initial gas limit.
-    /// @param _unsafeBlockSigner Initial unsafe block signer address.
-    /// @param _config            Initial resource config.
-    constructor(
-        address _owner,
-        uint256 _overhead,
-        uint256 _scalar,
-        bytes32 _batcherHash,
-        uint64 _gasLimit,
-        address _unsafeBlockSigner,
-        ResourceMetering.ResourceConfig memory _config
-    ) {
+    constructor() {
         initialize({
-            _owner: _owner,
-            _overhead: _overhead,
-            _scalar: _scalar,
-            _batcherHash: _batcherHash,
-            _gasLimit: _gasLimit,
-            _unsafeBlockSigner: _unsafeBlockSigner,
-            _config: _config
+            _owner: address(0xdEaD),
+            _overhead: 0,
+            _scalar: 0,
+            _batcherHash: bytes32(0),
+            _gasLimit: 1,
+            _unsafeBlockSigner: address(0),
+            _config: ResourceMetering.ResourceConfig({
+                maxResourceLimit: 1,
+                elasticityMultiplier: 1,
+                baseFeeMaxChangeDenominator: 2,
+                minimumBaseFee: 0,
+                systemTxMaxGas: 0,
+                maximumBaseFee: 0
+            }),
+            _batchInbox: address(0),
+            _addresses: SystemConfig.Addresses({
+                l1CrossDomainMessenger: address(0),
+                l1ERC721Bridge: address(0),
+                l1StandardBridge: address(0),
+                l2OutputOracle: address(0),
+                optimismPortal: address(0),
+                optimismMintableERC20Factory: address(0)
+            })
         });
+        Storage.setUint(START_BLOCK_SLOT, type(uint256).max);
```

Comment:
We should set the start block slot before calling initialize so that inside of initialize the setting of the start block is skipped. We should add a comment clarifying that its set particularly this way before initialize. Chains deployed before the introduction of the start block need to have the slot set manually and newly deployed chains will get the value set automatically